concourse/scripts/migrate_concourse_volume.sh

```diff
@@ -0,0 +1,377 @@
+    echo "Found existing persistent volume (/dev/sdf): ${found_origin_concourse_instance_id}"
+
+    if [ "${origin_concourse_instance_id}" != "${found_origin_concourse_instance_id}" ] || \
+        [ "${origin_concourse_volume_id}" != "${found_origin_concourse_volume_id}" ]; then
+        echo "Error: Instance ID or Volume ID in the state does not match with ones running."
+        echo "State file: ${BOSH_INIT_CONCOURSE_STATE}"
+        echo "State ids: ${origin_concourse_instance_id} ${origin_concourse_volume_id}"
+        echo "Running ids: ${found_origin_concourse_instance_id} ${found_origin_concourse_volume_id}"
+        exit 1
+    fi
+
+    echo "Terminating old concourse ${found_origin_concourse_instance_id}..."
+    terminate_instance "${found_origin_concourse_instance_id}"
+    wait_instance_state "${found_origin_concourse_instance_id}" "terminated"
+  else
+    echo "Unable to find a running bosh-init concourse. Has it been deleted already?"
+  fi
+}
+
+attach_old_volume_to_new_concourse() {
+  origin_concourse_instance_id="$(get_concourse_bosh_init_state_instance_id)"
+  echo "bosh-init state origin concourse instance ID: ${origin_concourse_instance_id}"
+
+  origin_concourse_volume_id="$(get_concourse_bosh_init_state_volume_id)"
+  echo "bosh-init state persistent volume: ${origin_concourse_volume_id}"
+
+  new_concourse_instance_id="$(find_all_instances_by_tags deployment:concourse job:concourse index:0 "director:${DEPLOY_ENV}")"
+  echo "New concourse instance ID: ${new_concourse_instance_id}"
+
+
+  new_concourse_volume_id="$(find_instance_disk_by_device_name "${new_concourse_instance_id}" "/dev/sdf")"
+  if [ "${new_concourse_volume_id}" ]; then
+    if [ "${new_concourse_volume_id}" == "${origin_concourse_volume_id}" ]; then
+      echo "The new concourse already has attached the volume ${origin_concourse_volume_id}. No need to update."
+    else
+      echo "Found existing persistent volume in new concourse (/dev/sdf): ${new_concourse_volume_id}"
+
+      echo "Stoping the new concourse instance ${new_concourse_instance_id}"
+      stop_instance "${new_concourse_instance_id}"
+      wait_instance_state "${new_concourse_instance_id}" "stopped"
+
+      echo "Detaching and deleting volume ${new_concourse_volume_id} from new concourse"
+      detach_volume "${new_concourse_volume_id}"
+      wait_volume_available "${new_concourse_volume_id}"
+      delete_volume "${new_concourse_volume_id}"
+
+      echo "Attaching ${origin_concourse_volume_id} to ${new_concourse_instance_id} as /dev/sdf"
+      attach_volume "${origin_concourse_volume_id}" "${new_concourse_instance_id}" "/dev/sdf"
+    fi
+  fi
+
+  echo "Starting again the new concourse instance ${new_concourse_instance_id}"
+  start_instance "${new_concourse_instance_id}"
+  wait_instance_state "${new_concourse_instance_id}" "running"
+
+  # TODO: Check if we need to do a monit restart all, not sure if it really starts :-m
```

Comment:
I'm not sure it's worth removing this comment as this script is going to be removed once the migration is complete.